takedown_feats.py

```diff
@@ -0,0 +1,421 @@
+
+            Args:
+                df (pd.DataFrame): The original dataframe containing fight data.
+
+            Returns:
+                pd.DataFrame: A dataframe with additional columns for significant strike features and differentials.
+            """
+            # Generate column names for takedowns and differentials
+            col_names = self.create_col_names_takedowns()
+            col_names_differential = self.create_col_names_differential_takedowns()
+
+            # Copy the input dataframe and calculate significant strike features
+            input_df = df.copy()
+            result_features = pd.DataFrame(columns=col_names)
+            result_features[col_names] = input_df.swifter.progress_bar(True).apply(lambda row: self.calculate_takedowns(input_df, row['fighter_a_id'], row['fighter_b_id'], row.name, col_names), axis=1)
+
+            # Combine the input dataframe with the calculated features
+            result_df = pd.concat([input_df, result_features], axis=1)
+
+            # Calculate and add differential features to the dataframe
+            result_df = self.calculate_takedowns_differential(result_df, col_names_differential)
+
+            return result_df
+
+    def calculate_takedowns(self, df, fighter_a_id, fighter_b_id, index, col_names):
+        """
+        Calculates significant strike features for both fighters in a match.
+
+        Args:
+            df (pd.DataFrame): The dataframe containing fight data.
+            fighter_a_id (str): The ID of fighter A.
+            fighter_b_id (str): The ID of fighter B.
+            index (int): The index of the current row in the dataframe.
+            col_names (list): List of column names to calculate.
+
+        Returns:
+            pd.Series: A series with calculated significant strike features.
+        """
+        # Initialize result list
+        res = []
+        fighter_a_last_3_fights, fighter_a_last_5_fights, fighter_a_prev_fights = self.get_fighter_fights(df, fighter_a_id, index)
+        fighter_b_last_3_fights, fighter_b_last_5_fights, fighter_b_prev_fights = self.get_fighter_fights(df, fighter_b_id, index)
+
+        for i in range(len(col_names)):
+            split = col_names[i].split('_')
+            fighter = split[0] # fighter-a, fighter-b
+            stat = split[1] # takedown-landed, takedown-accuracy, takedown-defense, takedown-absorbed
+            round = split[2]  # R1, R2, R3, R4, R5, overall
+            time_period = split[3] # last-3-fights, last-5-fights, alltime
+
+            # Get fights for the fighter
+            all_prev_fights = fighter_a_prev_fights if fighter == 'fighter-a' else fighter_b_prev_fights
+            last_3_fights = fighter_a_last_3_fights if fighter == 'fighter-a' else fighter_b_last_3_fights
+            last_5_fights = fighter_a_last_5_fights if fighter == 'fighter-a' else fighter_b_last_5_fights
+            
+            prev_fights = pd.DataFrame()
+
+            if fighter == 'fighter-a':
```

Comment:
You could replace the if statements with the statement below kind of how you did above to make the code more concise

`fighter_id = fighter_a_id if fighter == 'fighter-a' else fighter_b_id`